### PR TITLE
Bounding UDP payload size to 1200 and renaming

### PR DESCRIPTION
**Pull request by BiagioFesta**
As mentioned in #1154, it seems quinn sends UDP datagrams with a payload 1232 bytes by default.

This is easily reproducible with the client.rs/server.rs quinn example, making a GET request and serve a reasonable big file.
```
cargo run --example client https:/server:1234/README.md
```
```
[...]
Jul 06 02:59:45.746 TRACE drive{id=0}: quinn_proto::connection: sending 1232 bytes in 1 datagrams
[...]
```

---

Looking at the QUIC spec, ([quic-draft-32 #14](https://datatracker.ietf.org/doc/html/draft-ietf-quic-transport-34#section-14)) , it seems the protocol should supports networks where the maximum UDP payload is 1200 bytes.
Any larger value should be discovered using PMTUD.

This PR contains two changes:
* Renaming the [`const MIN_MTU: u16 = 1232;`](https://github.com/quinn-rs/quinn/blob/3f908a2c8c1ec4585212d776fafe536ea17bf2b4/quinn-proto/src/lib.rs#L313) into `MAX_INITIAL_UDP_PAYLOAD_SIZE`. This change has no impact from the semantic point of view. The rationale behind this change is because that value seems to be used across the code to compute what in QUIC is defined as "the maximum datagram size" (i.e., the UDP datagram payload).
* Changing this value from `1232` to `1200` and avoid quinn assembles packets (UDP datagrams) with a payload greater than this value.

### CODE REVIEW

File: quinn-proto/src/lib.rs [hunk 1]
```diff
@@ -310,7 +310,7 @@ const LOC_CID_COUNT: u64 = 8;
 const RESET_TOKEN_SIZE: usize = 16;
 const MAX_CID_SIZE: usize = 20;
 const MIN_INITIAL_SIZE: u16 = 1200;
-const MIN_MTU: u16 = 1232;
+const MAX_INITIAL_UDP_PAYLOAD_SIZE: u16 = 1200;
```
Comment (Ralith) [inline quinn-proto/src/lib.rs:313]:
```suggestion
const INITIAL_MAX_UDP_PAYLOAD_SIZE: u16 = 1200;
```

**Review by Ralith — CHANGES_REQUESTED** [review]
Thanks!

File: quinn-proto/src/lib.rs (hunk 1 above)
Comment (BiagioFesta) [reply quinn-proto/src/lib.rs]: Changed

**Comment by djc** [comment]
Just need to fix up the formatting a bit, then I'm happy to merge this.

**Comment by BiagioFesta** [comment]
> Just need to fix up the formatting a bit, then I'm happy to merge this.

I guess a `cargo fmt` should be enough even if I don't see a `rustfmt.toml`

**Comment by djc** [comment]
Would you mind squashing the formatting fixes into the first commit? Should be easy with an interactive rebase.

**Review by djc — APPROVED** [review]

**Pull request merged**